### PR TITLE
TiKV config: add abort-on-panic

### PR DESCRIPTION
**Pull request by yiwu-arbug**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

TiKV config: add abort-on-panic. The option is new in v5.0.2.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions) **version-specific, add version mark back**
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/tikv/tikv/pull/10164

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -11,6 +11,13 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 <!-- markdownlint-disable MD001 -->
 
+## 全局配置
+
+### abort-on-panic <span class="version-mark">从 v5.0.2 版本开始引入</span>
+
++ 设置 TiKV panic 时 abort 进程，允许系统生成 core dump。当这个配置项为 false 时，TiKV 使用 exit 退出进程。注意当把这个配置项打开以后，还需要进行相关的系统配置才能生成 core dump。建议把 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```
Comment (qiancai) [inline tikv-configuration-file.md:18]: 关于 “需要进行相关的系统配置”，请问这里具体是什么系统配置呢？

Comment (yiwu-arbug) [reply tikv-configuration-file.md:18]: 就是需要在系统中相应打开比如 ulimit -c 和配置 core dump 路径等。不同操作系统配置方式不一样，这里没有办法描述。

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -11,6 +11,13 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 <!-- markdownlint-disable MD001 -->
 
+## 全局配置
+
+### abort-on-panic <span class="version-mark">从 v5.0.2 版本开始引入</span>
```
Comment (TomShawn) [inline tikv-configuration-file.md:16]: 可以在 release-5.0 分支中加回来。因为 master 默认是最新版，具体版本号对读者意义不大。
```suggestion
### abort-on-panic
```

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by yiwu-arbug** [comment]
@TomShawn @qiancai ping

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -11,6 +11,13 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 <!-- markdownlint-disable MD001 -->
 
+## 全局配置
+
+### abort-on-panic
+
++ 设置 TiKV panic 时 abort 进程，允许系统生成 core dump。当这个配置项为 false 时，TiKV 使用 exit 退出进程。注意当把这个配置项打开以后，还需要进行相关的系统配置才能生成 core dump。建议把 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```
Comment (qiancai) [inline tikv-configuration-file.md:18]:
```suggestion
+ 设置 TiKV panic 时 abort 进程是否允许系统生成 core dump 文件。

    + 如果此配置项值为 false ，当 TiKV panic 时，TiKV 使用 exit 退出进程。
    + 如果此配置项值为 true ，当 TiKV panic 时，TiKV 允许系统在退出时生成 core dump 文件。要生成 core dump 文件，你还需要进行 core dump 相关的系统配置（比如打开 ulimit -c 和配置 core dump 路径，不同操作系统配置方式不同）。建议将 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -11,6 +11,17 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
...
 <!-- markdownlint-disable MD001 -->
 
+## 全局配置
+
+### abort-on-panic
+
++ 设置 TiKV panic 时 abort 进程是否允许系统生成 core dump 文件。
+
+    + 如果此配置项值为 false ，当 TiKV panic 时，TiKV 使用 exit 退出进程。
+    + 如果此配置项值为 true ，当 TiKV panic 时，TiKV 允许系统在退出时生成 core dump 文件。要生成 core dump 文件，你还需要进行 core dump 相关的系统配置（比如打开 ulimit -c 和配置 core dump 路径，不同操作系统配置方式不同）。建议将 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```
Comment (yiwu-arbug) [inline tikv-configuration-file.md:21]:
```suggestion
+ 设置 TiKV panic 时是否使用 abort 退出进程。此选项影响 TiKV 是否允许系统生成 core dump 文件。

    + 如果此配置项值为 false ，当 TiKV panic 时，TiKV 使用 exit 退出进程。
    + 如果此配置项值为 true ，当 TiKV panic 时，TiKV 使用 abort 退出进程。此时 TiKV 允许系统在退出时生成 core dump 文件。要生成 core dump 文件，你还需要进行 core dump 相关的系统配置（比如打开 ulimit -c 和配置 core dump 路径，不同操作系统配置方式不同）。建议将 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```

Comment (yiwu-arbug) [reply tikv-configuration-file.md:21]: @qiancai

Comment (qiancai) [reply tikv-configuration-file.md:21]: lgtm, please commit the changes.

**Comment by yiwu-arbug** [comment]
this one is ready to be merge, I think? @qiancai @TomShawn

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -11,6 +11,17 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 <!-- markdownlint-disable MD001 -->
 
+## 全局配置
+
+### abort-on-panic
+
++ 设置 TiKV panic 时是否使用 abort 退出进程。此选项影响 TiKV 是否允许系统生成 core dump 文件。
```
Comment (TomShawn) [inline tikv-configuration-file.md:18]:
```suggestion
+ 设置 TiKV panic 时是否使用 `abort` 函数退出进程。此选项影响 TiKV 是否允许系统生成 core dump 文件。
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -11,6 +11,17 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
...
+
+    + 如果此配置项值为 false ，当 TiKV panic 时，TiKV 使用 exit 退出进程。
```
Comment (TomShawn) [inline tikv-configuration-file.md:20]:
```suggestion
    + 如果此配置项值为 false ，当 TiKV panic 时，TiKV 使用 `exit` 函数退出进程。
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -11,6 +11,17 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
...
+    + 如果此配置项值为 true ，当 TiKV panic 时，TiKV 使用 abort 退出进程。此时 TiKV 允许系统在退出时生成 core dump 文件。要生成 core dump 文件，你还需要进行 core dump 相关的系统配置（比如打开 ulimit -c 和配置 core dump 路径，不同操作系统配置方式不同）。建议将 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```
Comment (TomShawn) [inline tikv-configuration-file.md:21]:
```suggestion
    + 如果此配置项值为 true ，当 TiKV panic 时，TiKV 使用 `abort` 函数退出进程。此时 TiKV 允许系统在退出时生成 core dump 文件。要生成 core dump 文件，你还需要进行 core dump 相关的系统配置（比如打开 `ulimit -c` 和配置 core dump 路径，不同操作系统配置方式不同）。建议将 core dump 生成路径设置在 TiKV 数据的不同磁盘分区，避免 core dump 文件占用磁盘空间过大，造成 TiKV 磁盘空间不足。
```

File: tikv-configuration-file.md (hunk 5 above)
Comment (yiwu-arbug) [reply tikv-configuration-file.md:18]: 改成 “`abort()` 调用” 和 "`exit()` 调用“ 这样的说法

Comment (TomShawn) [reply tikv-configuration-file.md:18]: Done。已改 comments

**Review by qiancai — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**